homebridge/thermostat.ts

```diff
@@ -0,0 +1,189 @@
+import { Logging, PlatformAccessory } from 'homebridge'
+
+import { RingDevice } from '../api'
+import { ThermostatMode } from '../api/ring-types'
+import { BaseDeviceAccessory } from './base-device-accessory'
+import { RingPlatformConfig } from './config'
+import { hap } from './hap'
+
+export class Thermostat extends BaseDeviceAccessory {
+  constructor(
+    public readonly device: RingDevice,
+    public readonly accessory: PlatformAccessory,
+    public readonly logger: Logging,
+    public readonly config: RingPlatformConfig
+  ) {
+    super()
+
+    const { Characteristic, Service } = hap
+
+    // Required Characteristics
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.CurrentHeatingCoolingState,
+      serviceType: Service.Thermostat,
+      getValue: async ({ setPoint, mode }) => {
+        if (mode === 'off') {
+          // The thermostat is set to 'off', so the thermostat is neither heating nor cooling
+          return Characteristic.CurrentHeatingCoolingState.OFF
+        }
+
+        const temperature = await this.getTemperatureFromComponentDevices()
+        if (!temperature || !setPoint) {
+          return
+        }
+        // Checking with a threshold to avoid floating point weirdness
+        const currentTemperatureEqualsTarget =
+            Math.abs(temperature - setPoint) < 0.1,
+          currentTemperatureIsHigherThanTarget = temperature - setPoint >= 0.1,
+          currentTemperatureIsLowerThanTarget = temperature - setPoint <= -0.1
+
+        if (currentTemperatureEqualsTarget) {
+          // The target temperature has been reached, so the thermostat is neither heating nor cooling
+          return Characteristic.CurrentHeatingCoolingState.OFF
+        }
+        if (currentTemperatureIsHigherThanTarget && mode === 'cool') {
+          // The current temperature is higher than the target temperature,
+          // and the thermostat is set to 'cool', so the thermostat is cooling
+          return Characteristic.CurrentHeatingCoolingState.COOL
+        }
+        if (
+          currentTemperatureIsLowerThanTarget &&
+          (mode === 'heat' || mode === 'aux')
+        ) {
+          // The current temperature is lower than the target temperature,
+          // and the thermostat is set to 'heat' or 'aux' (emergency heat), so the thermostat is heating
+          return Characteristic.CurrentHeatingCoolingState.HEAT
+        }
+        // The current temperature is either higher or lower than the target temperature,
+        // but the current thermostat mode would only to increase the difference,
```

Comment:
```suggestion
        // but the current thermostat mode would only increase the difference,
```

Reply:
Thanks! Committed in https://github.com/dgreif/ring/pull/331/commits/8559f7f44a73437ec23a1c4070dce19b6ea8f651